iml-agent/src/action_plugins/ldev.rs

```diff
@@ -0,0 +1,325 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{agent_error::ImlAgentError, env};
+use iml_wire_types::LdevEntry;
+use std::collections::BTreeSet;
+use tokio::{
+    fs::{File, OpenOptions},
+    io::{AsyncReadExt, AsyncWriteExt},
+};
+
+async fn write_to_file(content: String) -> Result<(), ImlAgentError> {
+    let ldev_path = env::get_ldev_conf();
+    let mut file = File::create(ldev_path).await?;
+    file.write_all(content.as_bytes()).await?;
+
+    Ok(())
+}
+
+async fn read_ldev_config() -> Result<String, ImlAgentError> {
+    let ldev_path = env::get_ldev_conf();
+
+    let mut file = OpenOptions::new()
+        .read(true)
+        .write(true)
```

Comment:
I added it based on the following documentation:
https://doc.rust-lang.org/std/fs/struct.OpenOptions.html#method.create
> Sets the option to create a new file, or open it if it already exists.
> In order for the file to be created, write or append access must be used.

Reply:
Hrrm, given this fn is called `read_ldev_config`, should it just check if the file exists and if so read it?

I think it could be surprising that this fn does a side-effect when it's called.

Reply:
It feels like an extra step to me but if you want I can add the additional check. I think the flow would be like this:
1. Check if file exists
   a. File exists
      i. Open the file and get a file handle
      i. Read the file contents into a String buffer
   b. File does not exist
      i. Return an empty string

In either case, the file will be created by the end of the action. Should I go ahead and make the change?

Reply:
I think it's worth doing. The file will only be completed by the end of the action if all the in-between code is successful.

As it stands now we have the (small) potential to have an empty file.

Reply:
Sounds good. I'll push an update shortly.

Reply:
These fns should help with that btw:

https://github.com/whamcloud/integrated-manager-for-lustre/blob/a86998683c14db89779510a675a517578bf2a792/iml-fs/src/lib.rs#L184-L192

https://github.com/whamcloud/integrated-manager-for-lustre/blob/a86998683c14db89779510a675a517578bf2a792/iml-fs/src/lib.rs#L69-L85

Reply:
I ended up using `read_to_string` (https://docs.rs/tokio/0.2.22/tokio/fs/fn.read_to_string.html) to avoid the mutable declaration.

Reply:
Let me know if this update looks good. It should follow the flow as discussed above.

Reply:
LGTM